examples/objectServerExample/src/main/java/io/realm/examples/objectserver/CounterActivity.kt

```diff
@@ -0,0 +1,178 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.objectserver
+
+import android.content.Intent
+import android.databinding.DataBindingUtil
+import android.graphics.PorterDuff
+import android.os.Bundle
+import android.support.annotation.ColorRes
+import android.support.v7.app.AppCompatActivity
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.TextView
+import io.realm.*
+import io.realm.examples.objectserver.databinding.ActivityCounterBinding
+import io.realm.examples.objectserver.model.CRDTCounter
+import io.realm.kotlin.syncSession
+import io.realm.kotlin.where
+import me.zhanghai.android.materialprogressbar.MaterialProgressBar
+import java.util.*
+import java.util.concurrent.atomic.AtomicBoolean
+
+class CounterActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityCounterBinding
+
+    private val downloadListener = ProgressListener { progress ->
+        downloadingChanges.set(!progress.isTransferComplete)
+        runOnUiThread(updateProgressBar)
+    }
+    private val uploadListener = ProgressListener { progress ->
+        uploadingChanges.set(!progress.isTransferComplete)
+        runOnUiThread(updateProgressBar)
+    }
+    private val updateProgressBar = Runnable { updateProgressBar(downloadingChanges.get(), uploadingChanges.get()) }
+
+    private val downloadingChanges = AtomicBoolean(false)
+    private val uploadingChanges = AtomicBoolean(false)
+
+    private lateinit var realm: Realm
+    private lateinit var session: SyncSession
+    private var user: SyncUser? = null
+
+    private lateinit var counterView: TextView
+    private lateinit var progressBar: MaterialProgressBar
+    private lateinit var counters: RealmResults<CRDTCounter> // Keep strong reference to counter to keep change listeners alive.
+
+    private val loggedInUser: SyncUser?
+        get() {
+            var user: SyncUser? = null
+
+            try {
+                user = SyncUser.current()
+            } catch (ignore: IllegalStateException) {
+            }
+
+            if (user == null) {
+                startActivity(Intent(this, LoginActivity::class.java))
+            }
+
+            return user
+        }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_counter)
+        counterView = binding.textCounter
+        progressBar = binding.progressbar
+        binding.upper.setOnClickListener { adjustCounter(1) }
+        binding.lower.setOnClickListener { adjustCounter(-1) }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        user = loggedInUser
+        user?.let {
+
+            // Create a RealmConfiguration for our user
+            val config = user!!.createConfiguration(BuildConfig.REALM_URL)
```

Comment:
prefer `val user = loggedInUser; if(user != null) {` to have safe-casting.

Reply:
`it` is actually safe here.